common/nation.cpp

```diff
@@ -455,166 +455,80 @@ struct nation_type *nation_by_number(const Nation_type_id nation)
   if (nation < 0 || nation >= game.control.nation_count) {
     return nullptr;
   }
-  return nations + nation;
+  return &nations[0] + nation;
```

Comment:
`&nations[nation]`

Reply:
:smile:  Looking at C too long makes me forget the simplest of things xD